### DISCUSSION TITLE
Copying hero 10 settings to another hero 10 via QR code

### DISCUSSION THREAD

**Discussion by f3rn4ndrum5** [gopro/labs#332] [Q&A]
Hello!
Can I make a one of my Hero10 show all it's settings via QR code in either  screen so I can scan it with the another Hero10?

Like presets, date, time, configuration etc?

Thanks in advance!

f3

**Comment by dnewman-gpsw** [comment]
This was something we tried and demo'd in the original hackathon on HERO5 (see the fun video we made https://www.instagram.com/p/CAv08o6j_fc/)

The problem is the GoPro screen is small, and the near focus for another GoPro seeing a GoPro screen is about 30cm/1foot.  This means only low density QR Codes can be used, so not many settings can be transferred.  As the screen got bigger in later models, the camera added presets and more settings, so the problem persists today. GoPro cameras have a lot of settings.

**Reply by f3rn4ndrum5** [reply]
kk tnx.

Like, following up my post on https://github.com/gopro/labs/discussions/310 - two hero10s with "the same settings" (cameras were restored to factory and even !RESET via QR and precision time set, no gps, no wifi) both behave difrently using the same https://gopro.github.io/labs/control/construction/ (against your recommendation) after a couple of weeks of testing using the suggested qr (https://gopro.github.io/labs/control/dailytl/) 

I have found out that the construction timelapse it's just more reliable and can be set to work at any moment. the DailyTL waits until the window opens to start working and the camera stays on all day and all three (h11, h10, h10) get warm on the caribean heat.

So, still, one h10 its a champ while the other it's kind of wonky despite having "the same everyting" in regards of settings etc. hence my question

The H11 does whatever it wants (un either  construction or daylytl). Sometimes it starts doing it's thing and then it says next picture will be taken in 18hours... so. kind o f crazy.

I'm testing the h11 with a different USB adaptor with more than the suggested 2A and taking notes because the adaptors seem to me are a crucial part in this. A 3A usb adaptor eliminates the bootloob stated in my origianl post.

thanks again for your answer

f3

**Reply by dnewman-gpsw** [reply]
Yes, construction timelapse is flakey, all cameras, just random, and I have never determine why.  Some particular cameras seem more reliable, again nothing to do a the camera model, or their settings. Cameras getting warm for timelapse should not be a problem, are you saying they are shutting down for overheating? 

So for camera with internal battery, the construction script is the best. The reliability with the internal battery is also better.  You just need to replace the battery every week or every few days, based on your settings.

For external power dailytl is better, more constantly reliable.

Best overall, use an external timer switch USB power on and off, and wake of power triggers with the GoPro and a boot command. Now the reliable issue will be external to the GoPro as the WAKE on power is 100% consistent.

**Reply by f3rn4ndrum5** [reply]
cameras are not shutting down, they are just getting very warm when uning dailytl. they are "on" all day.

in my other post you said "that type of script is no longer recommended as it is not reliable -- too many wake/shutdowns per day."  - what's the concern with the wake/shutdowns  cycle? 

the site has ac current but it fails daily. so an energy efficient script it's better imo

tnx

f3

**Reply by dnewman-gpsw** [reply]
Warm is normal for GoPro, that is no concern, particularly with no battery.

Issue with wakes, it doesn't always wake.  Not harm to the camera, just a messed up timelapse. The hardware wake timer was not designed for this purpose, Labs uses it far more aggressively than the on-camera schedule capture (which is only once, not even once per day.)  So if the wake timer approximately reliable 299 out of 300 starts, schedule capture will be feel very reliable, but taking 60 photos a day, the script will fail within the week.  Schedule capture might even be 100% reliable, as Labs is simply more hacky, resulting in wake start being less than 100% reliable.

**Comment by cprima** [comment]
> Hello! Can I make a one of my Hero10 show all it's settings via QR code in either screen so I can scan it with the another Hero10?
> 
> Like presets, date, time, configuration etc?
> 

As a user I am evaluating GoPro Labs since a few weeks, and decided that I go a "QR-Code first" route:
I simply don't make many settings manually,
but print out or store QR codes.

I store the strings in a spreadsheet so that I do not need to use the app.

![image](https://user-images.githubusercontent.com/943871/236200433-3a6f0ec6-5ed8-4b27-9835-bae1ee7ea1f4.png)

![IMG_20230425_141935](https://user-images.githubusercontent.com/943871/234276734-58efa550-a2d3-482c-b176-fa14ab46681e.jpg)

![image](https://user-images.githubusercontent.com/943871/236199229-2bca76e1-f487-4021-9881-1b273d64b038.png)


I know this is not directly answering the request, but as I am very happy with the workflow I wanted share it as a workaround for the intention of the changerequest.

A few settings need to be made manually, IIRC AntiFlicker.


And then there is the metadata extractor, which I looked breifly into if I could reverse-extract input for a QR code from
https://gopro.github.io/labs/control/metadata/
But that being "only" metadata I did not further pursue the idea.


I think for a user settled into the Labs firmware a QR-code first approach makes the most sense:
I do not even want to "waste the time" setting up the first camera manually…

**Reply by dnewman-gpsw** [reply]
Note:  AntiFlicker does not need to be set manually. Just set a 50Hz frame rate (p25,p50, p100 or p200) via QR Code, and it will auto switch to 50Hz antiflicker. The reverse for 60Hz.

**Reply by f3rn4ndrum5** [reply]
> > Hello! Can I make a one of my Hero10 show all it's settings via QR code in either screen so I can scan it with the another Hero10?
> > Like presets, date, time, configuration etc?
> 
> As a user I am evaluating GoPro Labs since a few weeks, and decided that I go a "QR-Code first" route: I simply don't make many settings manually, but print out or store QR codes.
> 
> I store the strings in a spreadsheet so that I do not need to use the app.
> 
> ![image](https://user-images.githubusercontent.com/943871/236200433-3a6f0ec6-5ed8-4b27-9835-bae1ee7ea1f4.png)
> 
> ![IMG_20230425_141935](https://user-images.githubusercontent.com/943871/234276734-58efa550-a2d3-482c-b176-fa14ab46681e.jpg)
> 
> ![image](https://user-images.githubusercontent.com/943871/236199229-2bca76e1-f487-4021-9881-1b273d64b038.png)
> 
> I know this is not directly answering the request, but as I am very happy with the workflow I wanted share it as a workaround for the intention of the changerequest.
> 
> A few settings need to be made manually, IIRC AntiFlicker.
> 
> And then there is the metadata extractor, which I looked breifly into if I could reverse-extract input for a QR code from https://gopro.github.io/labs/control/metadata/ But that being "only" metadata I did not further pursue the idea.
> 
> I think for a user settled into the Labs firmware a QR-code first approach makes the most sense: I do not even want to "waste the time" setting up the first camera manually…

This works, I have done it and also have a few qrcode cards in every rig we built (pelican case with gopro with a UV filter and attachments for grip). So when we go and download the media we have the cards in the box to restore the timelapse.
As for setting the camera without using the menu, I'll check it out to see if it's viable for our case.

As I said before, we have one h10 already installed and has been flawless, of the two I have here, one is a champ and the other one was kind of wonky but it's working fine since a couple of days ago after I used the !Reset and formated the SD card.

The H11 is like a cat. It does whatever it wants in any mode or setting and stops or starts whenever it feels like it... disappointed.gif

Still it would be very usefull, imo, to implement this "share settings via QRCode or app or bluetooth or whatever" so you can replicate the exact same settings from camera to camera in a simple step.
Even creating a .jpg file into the sd card with the more complex qrcode needed if the screen is to small/lowres.

Thanks for your contribution!

**Reply by cprima** [reply]
Thanks @dnewman-gpsw  for the heads up in the antiflicker! Mich appreciated!

@f3rn4ndrum5 Writing a jpg is a nice idea! If connectivity with the web can be assumed even just the long string would be helpful that could be used on / passed on to https://gopro.github.io/labs/control/set/cmd=... would be great. For example if there was !WRITEOUT which appends to qrlog.txt and with a SD card reader a browser page could be opened

(Although that would only represent a specific moment in time, and not be representing a clip/picture -- settings and presets might change "any time" )
